### ISSUE TITLE
ChronicleHashCloseOnExitHook is causing more harm than good

### ISSUE THREAD

**Issue by udaysagar2177** [OpenHFT/Chronicle-Map#178]
Hi,

**Requirement:**
We want to control the JVM shutdown process so that we can do certain pre-shutdown actions like transform or backup the data etc. Our pre-shutdown actions involves reading the current Chronicle maps or creating new maps. 

**Problem:**
Since ChronicleHashCloseOnExitHook is a JVM Shutdown Hook, the `closeAll()` call is triggered immediately after JVM receives a shutdown signal and that sets `ChronicleHashCloseOnExitHook.maps` to `null` preventing any more map creations.

**Proposed Solution:**
I think we need a handle on whether ChronicleHashCloseOnExitHook does anything. Since we are careful about closing the maps, we don't want a pre-configured shutdown hook interrupting our work. The easiest way is to add a user configurable flag `skipCloseOnExitHook` on ChronicleHashBuilder that controls whether to register the created ChronicleMap into the shutdownHook. We could use the flag something like below. Let me know if you would like a PR for this.

ChronicleMapBuilder.java
```
    private void prepareMapPublication(VanillaChronicleMap map) throws IOException {
        establishReplication(map);
        map.setResourcesName();
        map.registerCleaner();
        // Ensure safe publication of the ChronicleMap
        OS.memory().storeFence();
        if (!skipCloseOnExitHook) {
            map.addToOnExitHook();
        }
    }
```

**Comment by udaysagar2177** [comment]
@leventov Could you please take a look at this?

**Comment by leventov** [comment]
Looks reasonable, but I'm not the maintainer of this repo

**Comment by udaysagar2177** [comment]
@leventov Thanks!

@RobAustin / @peter-lawrey Would you mind looking at this issue?

**Comment by dpisklov** [comment]
@udaysagar2177 PR would be good, with default behaviour unchanged

**Comment by hft-team-city** [comment]
Released in [Chronicle-Map-3.17.3](https://github.com/OpenHFT/Chronicle-Map/releases/chronicle-map-3.17.3), [BOM-2.17.262](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.262)
